demo/campaign/src/common/blocks/CommonImageBlock.tsx

```diff
@@ -0,0 +1,78 @@
+import { calculateInheritAspectRatio, generateImageUrl } from "@comet/cms-site";
+import { MjmlImage, MjmlStyle } from "@luma-team/mjml-react";
+import { PixelImageBlockData } from "@src/blocks.generated";
+import { css } from "@src/util/stylesHelper";
+import { theme } from "@src/util/theme";
+import * as React from "react";
+
+import { getDamAllowedImageWidth } from "../helpers/imageBlockHelpers";
+
+interface Props extends React.ComponentProps<typeof MjmlImage> {
+    data: PixelImageBlockData;
+    desktopRenderWidth: number;
+    growToOverflowWidth?: boolean;
+}
+
+export const commonImageBlockStyles = (
+    <MjmlStyle>{css`
+        @media (max-width: ${theme.mailSize.contentWidth - 1}px) {
+            .image-block > table > tbody > tr > td {
+                width: 100% !important;
+            }
+
+            .image-block img {
+                height: auto !important;
+            }
+        }
+
+        @media (min-width: ${theme.mailSize.contentWidth + 1}px) {
+            .image-block--grow-to-overflow-width > table > tbody > tr > td {
+                width: 100% !important;
+            }
+
+            .image-block--grow-to-overflow-width img {
+                height: auto !important;
+            }
+        }
+    `}</MjmlStyle>
+);
+
+export const CommonImageBlock = ({ data, desktopRenderWidth, growToOverflowWidth, ...restProps }: Props) => {
+    const { damFile, cropArea, urlTemplate } = data;
+
+    if (!damFile?.image) {
+        return null;
+    }
+
+    const usedCropArea = cropArea ?? damFile.image.cropArea;
+    const usedAspectRatio = calculateInheritAspectRatio(damFile.image, usedCropArea);
+
+    const imageUrl: string = generateImageUrl(
+        {
+            width: getDamAllowedImageWidth(desktopRenderWidth),
+            src: urlTemplate,
+        },
+        usedAspectRatio,
+    );
```

Comment:
@jamesricky I added 1020px to the allowedImageSizes as discussed.

I also tried to cover the case that another image size is provided by checking, which validWidth is approximately twice the width given to avoid that images are not displayed at all. Of course, adding the exact value would still be the better solution.